packages/types/src/pnum.ts

```diff
@@ -0,0 +1,115 @@
+import { BigNumber } from 'bignumber.js';
+import { round } from '@penumbra-zone/types/round';
+import { LoHi, joinLoHi, splitLoHi } from '@penumbra-zone/types/lo-hi';
+import { Amount } from '@penumbra-zone/protobuf/penumbra/core/num/v1/num_pb';
+import { ValueView } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { getAmount, getDisplayDenomExponentFromValueView } from '@penumbra-zone/getters/value-view';
+import { removeTrailingZeros } from '@penumbra-zone/types/shortify';
+
+/**
+ * pnum (penumbra number)
+ *
+ * In Penumbra a number can be in the form of a base unit (bigint, LoHi, Amount, ValueView)
+ * or a number with decimals for display purposes (string, number)
+ *
+ * This function handles all these cases automatically internally
+ * - when input is a bigint, LoHi, Amount, or ValueView, it is assumed to be in base units
+ * - when input is a string or number, it is multiplied by 10^exponent and converted to base units
+ *
+ * Likewise for all methods, the outputs are
+ * - in base units for bigint, LoHi, Amount and ValueView
+ * - in display form with decimals for string and number
+ *
+ * @param input
+ * @param exponent
+ */
+function pnum(
+  input: string | number | LoHi | bigint | Amount | ValueView | undefined,
+  exponentInput = 0,
+) {
```

Comment:
issue: this feels very class-like (private vars, exposed methods, constructor like initialization). Think we should convert it.

Reply:
I did this on purpose as it's shorter: `pnum(val)` vs `new PNum(val)`. My guess is that we'll use it quite often and this way it looks cleaner imo, especially if we need to do it inline inside a react component.